bundles/org.openhab.binding.somfymylink/src/main/java/org/openhab/binding/somfymylink/internal/handler/SomfyMyLinkBridgeHandler.java

```diff
@@ -0,0 +1,549 @@
+
+    private @Nullable ExecutorService commandExecutor;
+
+    // Gson & parser
+    private final Gson gson = new Gson();
+
+    public SomfyMyLinkBridgeHandler(Bridge bridge,
+            @Nullable SomfyMyLinkStateDescriptionOptionsProvider stateDescriptionProvider) {
+        super(bridge);
+
+        this.discovery = new SomfyMyLinkDeviceDiscoveryService(this);
+        this.stateDescriptionProvider = stateDescriptionProvider;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Command received on mylink {}", command);
+
+        try {
+            if (CHANNEL_SCENES.equals(channelUID.getId())) {
+                if (command instanceof RefreshType) {
+                    return;
+                }
+
+                if (command instanceof StringType) {
+                    Integer sceneId = Integer.decode(command.toString());
+                    commandScene(sceneId);
+                }
+            }
+        } catch (SomfyMyLinkException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing mylink");
+        config = getThing().getConfiguration().as(SomfyMyLinkConfiguration.class);
+
+        commandExecutor = Executors.newSingleThreadExecutor(new NamedThreadFactory(thing.getUID().getAsString(), true));
+
+        if (validConfiguration(config)) {
+            this.discoveryServiceRegistration = FrameworkUtil.getBundle(SomfyMyLinkBridgeHandler.class)
+                    .getBundleContext().registerService(DiscoveryService.class, this.discovery, null);
+            this.discovery.activate(null);
+
+            // kick off the bridge connection process
+            this.scheduler.execute(this::connect);
+        }
+    }
+
+    private boolean validConfiguration(@Nullable SomfyMyLinkConfiguration config) {
+        if (config == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "mylink configuration missing");
+            return false;
+        }
+
+        if (StringUtils.isEmpty(config.ipAddress)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "mylink address not specified");
+            return false;
+        }
+
+        return true;
+    }
+
+    private void connect() {
+        try {
+            SomfyMyLinkConfiguration config = this.config;
+
+            if (config == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "mylink config not specified");
+                return;
+            }
+
+            if (StringUtils.isEmpty(config.ipAddress) || StringUtils.isEmpty(config.systemId)) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "mylink config not specified");
+                return;
+            }
```

Comment:
These configuration checks should be done in the initialize() method.

Reply:
done